markdown.go

```diff
@@ -343,45 +343,45 @@ func WithNoExtensions() Option {
 // the override function indicates an override did not occur, the refids at
 // the bottom will be used to fill in the link details.
 func WithRefOverride(o ReferenceOverrideFunc) Option {
-	return func(p *Processor) {
+	return func(p *Markdown) {
 		p.referenceOverride = o
 	}
 }
 
-// Markdown is the main entry point to Blackfriday. It parses and renders a
+// Run is the main entry point to Blackfriday. It parses and renders a
 // block of markdown-encoded text.
 //
-// The simplest invocation of Markdown takes one argument, input:
-//     output := Markdown(input)
+// The simplest invocation of Run takes one argument, input:
+//     output := Run(input)
 // This will parse the input with CommonExtensions enabled and render it with
 // the default HTMLRenderer (with CommonHTMLFlags).
 //
-// Variadic arguments opts can customize the default behavior. Since Processor
+// Variadic arguments opts can customize the default behavior. Since Markdown
 // type does not contain exported fields, you can not use it directly. Instead,
 // use the With* functions. For example, this will call the most basic
 // functionality, with no extensions:
-//     output := Markdown(input, WithNoExtensions())
+//     output := Run(input, WithNoExtensions())
 //
 // You can use any number of With* arguments, even contradicting ones. They
 // will be applied in order of appearance and the latter will override the
 // former:
-//     output := Markdown(input, WithNoExtensions(), WithExtensions(exts),
+//     output := Run(input, WithNoExtensions(), WithExtensions(exts),
 //         WithRenderer(yourRenderer))
-func Markdown(input []byte, opts ...Option) []byte {
+func Run(input []byte, opts ...Option) []byte {
 	r := NewHTMLRenderer(HTMLRendererParameters{
 		Flags: CommonHTMLFlags,
 	})
 	optList := []Option{WithRenderer(r), WithExtensions(CommonExtensions)}
 	optList = append(optList, opts...)
-	parser := NewProcessor(optList...)
+	parser := New(optList...)
```

Comment:
I believe the rationale behind this design choice was "Minimize implicit allocations."  If your suggested syntax had been allowed Go would have had to allocate a slice behind the scenes.